src/mscorlib/shared/System/Diagnostics/StackFrame.cs

```diff
@@ -0,0 +1,272 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System.Text;
+using System;
+using System.IO;
+using System.Reflection;
+using System.Diagnostics.Contracts;
+
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// There is no good reason for the methods of this class to be virtual.
+    /// </summary>
+    public partial class StackFrame
+    {
+        private MethodBase method;
```

Comment:
Since this class is not serializable and you're making other changes besides just splitting this up, we might also consider renaming the fields to conform to the CoreFX coding guidelines (prefix with underscores and maybe also omit the Hungarian notation).